### PR TITLE
Add a no-ttl flag to etcdctl migrate to discard keys on transform.

### PR DESCRIPTION
**Pull request by timothysc**
Fixes: #6767 

Adds a simple no-ttl flag to allow users to skip any ttl keys.

### CODE REVIEW

**Comment by xiang90** [comment]
@hongchaodeng Can you please review this?

File: etcdctl/ctlv3/command/migrate_command.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ import (
 )
 
 var (
+	migrateNoTTL       bool
```
Comment (hongchaodeng) [inline etcdctl/ctlv3/command/migrate_command.go]: migrateExcludeTTLKey

File: etcdctl/ctlv3/command/migrate_command.go [hunk 2]
```diff
@@ -55,6 +56,7 @@ func NewMigrateCommand() *cobra.Command {
 		Run:   migrateCommandFunc,
 	}
 
+	mc.Flags().BoolVar(&migrateNoTTL, "no-ttl", false, "Do not convert TTL keys")
```
Comment (hongchaodeng) [inline etcdctl/ctlv3/command/migrate_command.go]: I'm fine with the flag name.
Can you change the var to "migrateExcludeTTLKey"?

Comment (timothysc) [inline etcdctl/ctlv3/command/migrate_command.go]: Sure.

File: etcdctl/ctlv3/command/migrate_command.go [hunk 3]
```diff
@@ -311,6 +313,10 @@ func defaultTransformer() (io.WriteCloser, io.ReadCloser, chan error) {
 				return
 			}
 
+			if migrateExcludeTTLKey && node.TTL != 0 {
```
Comment (xiang90) [inline etcdctl/ctlv3/command/migrate_command.go]: we should move this to writeKey func, which emits the keys to transformer. or this flag only works for the default transformer.

**Comment by xiang90** [comment]
Please also change the commit message to ctlv3: Add a no-ttl flag to etcdctl migrate to discard keys on transform.

Or CI wont pass.

**Comment by xiang90** [comment]
LGTM. Will merge when CI turns green.

**Comment by wojtek-t** [comment]
@xiang90 - it seems that test passed

**Pull request merged**